pandas/core/indexes/interval.py

```diff
@@ -1090,6 +1091,36 @@ def equals(self, other):
     def overlaps(self, other):
         return self._data.overlaps(other)
 
+    def intersection(self, other, sort=False):
```

Comment:
does this need a @appender for the doc-string?

Reply:
the dict `_interval_shared_docs` in `IntervalArray` and the dict `_index_shared_docs` in `Index` don't have the key `intersection`.

Reply:
so this would take on Index doc-string. Pls check this; you may need to add it to the shared_docs to have a more customized doc-string.

Reply:
Confirmed that this picks up the Index.intersection. Can customize for IntervalIndex down the road.